### PR TITLE
Add quote reporting feature

### PR DESCRIPTION
**Pull request by Bruception**
### Description

I have designed the reporting system to be extensible/generic. Such that if we want to support reporting for another feature on the site, we can easily set that up.

closes: #1158

### CODE REVIEW

**Comment by Miodec** [comment]
I dont think going through the bot is necessary - the server itself could just create the issue.

Also im thinking if automating issues through user reports is a good idea. What if someone spams the API and we potentially get blacklisted from the Github API?

Honestly i was thinking of something way more simple where its just stored in a reports collectoin and i can look at it manually / through my admin panel from time to time and address the reports.

**Comment by Bruception** [comment]
> I dont think going through the bot is necessary - the server itself could just create the issue.
> 
> Also im thinking if automating issues through user reports is a good idea. What if someone spams the API and we potentially get blacklisted from the Github API?
> 
> Honestly i was thinking of something way more simple where its just stored in a reports collectoin and i can look at it manually / through my admin panel from time to time and address the reports.

I agree. I was just following your idea from here: https://github.com/Miodec/monkeytype/issues/1158#issuecomment-997921454

**Comment by Miodec** [comment]
Wow.. Did i really say that... Yeah lets not do what i said 😂

**Comment by Bruception** [comment]
Demonstration of the popups for the report quote feature.

Screen Shot 2022-01-30 at 12 26 48 AM
Screen Shot 2022-01-30 at 12 27 00 AM

**Comment by Miodec** [comment]
Dont forget to also add it to the `rate quote` popup.

**Comment by Bruception** [comment]
> Dont forget to also add it to the `rate quote` popup.

I'm having an issue where when the report popup is shown and I try to type into the comment textarea. The TestUI is triggered. I commented [this line](https://github.com/Miodec/monkeytype/blob/master/src/js/popups/quote-search-popup.js#L115) too.

**Comment by Miodec** [comment]
Did removing that fix it?

**Comment by Bruception** [comment]
> Did removing that fix it?

No

**Comment by Miodec** [comment]
discord.gg/monkeytype, #development channel.

File: src/js/popups/quote-report-popup.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    const { quotes } = await Misc.getQuotes(Config.language);
+    state.quoteToReport = quotes.find((quote) => {
+      return quote.id === quoteId;
+    });
+
+    $("#quoteBeingReported").text(state.quoteToReport.text);
+    $("#quoteReportReason").val("Grammatical error");
+    $("#quoteReportComment").val("");
+    $("#quoteReportPopupWrapper .characterCount").text("-");
+    grecaptcha.reset();
```
Comment (Bruception) [inline src/js/popups/quote-report-popup.js:33]: This isn't resetting the captcha. Any additional steps?

**Review by Miodec — APPROVED** [review]
Thanks a lot!

**Pull request merged**